### PR TITLE
Explore Metrics: Add button to add to exploration

### PR DESCRIPTION
**Pull request by svennergr**
## What is this feature?

Adds a PoC implementation of the `Add to Exploration` button to Explore Metrics.

## Demo

https://github.com/user-attachments/assets/c92f2b71-73ec-4c10-9685-9b1b180d8fc3

## How does it work?

When the [Investigations App](https://github.com/grafana/investigations-app) is installed, it  uses a [UI extension](https://grafana.com/developers/plugin-tools/how-to-guides/ui-extensions) to connect with the new `AddToExplorationsButton`. This button appears in various panels throughout Explore Metrics, allowing users to add metrics panels to their exploration (alongside logs panels, etc.).

## Notes for reviewers

Running this locally requires setup in both `grafana/grafana` and `grafana/investigations-app`.

### `grafana/investigations-app`

1. Clone [grafana/investigations-app](https://github.com/grafana/investigations-app) and switch to its `svennergr/add-explore-metrics-link` branch.
2. Install frontend dependencies (`npm install`)
3. Build the backend with `mage -v`
4. Build the frontend with the `build` script (`npm run build`)

### `grafana/grafana`

1. Check out the `svennergr/explorations-explore-metrics` branch.
2. Create a symlink for `grafana-explorations-app` in the Grafana plugins directory:
```
cd grafana/data/plugins/
ln -s /ABSOLUTE/PATH/TO/investigations-app/dist grafana-explorations-app
```
3. Configure Grafana to allow unsigned plugins by enabling `development` mode for apps in `conf/custom.ini`:
```
app_mode = development
```
4. Start/restart grafana

After following these steps, you should see the `AddToExplorationsButton` in panels throughout Explore Metrics:

demo explorations button

### CODE REVIEW

**Comment by svennergr** [comment]
Hi @NWRichmond, I saw you enabled auto-merge. We would still need an approval for this to be merged 😆

**Comment by bohandley** [comment]
I think I need some help with the instructions for setting this up. I am able to build the investigations app. After creating a symlink from the investigations-app to the "grafana-explorations-app", I am able to see the explorations app in the list of plugins. I do not see the additional UI (the button) in Explore metrics though :(. 

I see that I can enable the "grafana-explorations-app" in the plugins list. Clicking on that app, I see is needs to be enabled. After enabling it, the new UI still does not show up.

Additional question: Why does this have to be symlinked to a new name? investigations-app => grafana-explorations-app? Can't we just build this in our plugins directory? My plugin directory is not in grafana/data/plugins. I created a new directory and changed the directory for my plugins to `plugins = ./data/plugins`

**Comment by svennergr** [comment]
> I see that I can enable the "grafana-explorations-app" in the plugins list. Clicking on that app, I see is needs to be enabled. After enabling it, the new UI still does not show up.

Yea, sorry, Brendan - I need to work on the readme for the investigations app. I think the only part you are missing is enabling the `appSidecar` feature flag. Could you try that?

> Additional question: Why does this have to be symlinked to a new name? investigations-app => grafana-explorations-app? Can't we just build this in our plugins directory? My plugin directory is not in grafana/data/plugins. I created a new directory and changed the directory for my plugins to plugins = ./data/plugins

Yea, the name isn't really needed. There were just some discussions about the name in the beginning. You can also install/build it directly in your plugins directory.

**Comment by bohandley** [comment]
Woohoo! Adding the  feature toggle `appSidecar` was the thing that fixed it. Thank you @svennergr!

File: public/app/features/trails/MetricSelect/AddToExplorationsButton.tsx [hunk 1]
```diff
@@ -0,0 +1,139 @@
+import { DataFrame, TimeRange } from '@grafana/data';
+import { usePluginLinks } from '@grafana/runtime';
+import { SceneComponentProps, sceneGraph, SceneObjectBase, SceneObjectState, SceneQueryRunner } from '@grafana/scenes';
+import { DataQuery, DataSourceRef } from '@grafana/schema';
+import { IconButton } from '@grafana/ui';
+
+import MimirLogo from '../../../plugins/datasource/prometheus/img/mimir_logo.svg';
```
Comment (bohandley) [inline public/app/features/trails/MetricSelect/AddToExplorationsButton.tsx:7]: Would be nice to have this not coupled to Grafana :)

File: public/app/features/trails/MetricSelect/AddToExplorationsButton.tsx [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+      this.setState({ context: ctx });
+    }
+  };
+
+  public static Component = ({ model }: SceneComponentProps<AddToExplorationButton>) => {
+    const { context, disabledLinks } = model.useState();
+    const { links } = usePluginLinks({ extensionPointId, context, limitPerPlugin: 1 });
+    const link = links.find((link) => link.pluginId === explorationsPluginId);
+
+    if (!link) {
```
Comment (bohandley) [inline public/app/features/trails/MetricSelect/AddToExplorationsButton.tsx:105]: This part confused me when I had not added the sidecar feature toggle. Would it be possible to wrap the component in this and add some comments as documentation? 

```
// This feature is part of the explorations app. Here is a link to the repo. The repo has instructions for building the app and the necessary feature toggles. LINK
```

But I can see how having the check inside the component is better for tests.

**Review by bohandley — APPROVED** [review]
Just a few concerns on the set up instructions and comments that would be helpful in the code. I think it is wise to be more verbose on new integrations earlier on. Like, what is a pluginLInk and why don't I see it anywhere else in Explore metrics?

But other than that, this looks really cool!

**Comment by NWRichmond** [comment]
> Hi @NWRichmond, I saw you enabled auto-merge. We would still need an approval for this to be merged 😆

For sure - I was OOO yesterday, and was hoping this could be merged as soon as O11y Metrics had time to review :)

File: public/app/features/trails/MetricSelect/AddToExplorationsButton.tsx (hunk 1 above)
Comment (NWRichmond) [reply public/app/features/trails/MetricSelect/AddToExplorationsButton.tsx:7]: Totally, this is one more thing that will need decoupling before Explore Metrics is externalized.

**Pull request merged**